src/coreComponents/constitutive/solid/PorousSolid.hpp

```diff
@@ -55,25 +55,37 @@ class PorousSolidUpdates : public CoupledSolidUpdates< SOLID_TYPE, BiotPorosity,
   GEOSX_HOST_DEVICE
   void smallStrainUpdate( localIndex const k,
                           localIndex const q,
+                          real64 const & pressure,
                           real64 const & deltaPressure,
                           real64 const ( &strainIncrement )[6],
-                          real64 ( & stress )[6],
+                          real64 ( & totalStress )[6],
                           real64 & dPorosity_dPressure,
                           real64 & dPorosity_dVolStrain,
-                          real64 & dTotalStress_dPressure,
+                          real64 ( & dTotalStress_dPressure )[6],
                           DiscretizationOps & stiffness ) const
   {
-    m_solidUpdate.smallStrainUpdate( k, q, strainIncrement, stress, stiffness );
+    // Compute effective stress and store in totalStress
+    m_solidUpdate.smallStrainUpdate( k, q, strainIncrement, totalStress, stiffness );
 
     updateBiotCoefficient( k );
 
+    // Compute  total stress
+    real64 const biotCoefficient = m_porosityUpdate.getBiotCoefficient( k );
+    LvArray::tensorOps::symAddIdentity< 3 >( totalStress, -biotCoefficient * ( pressure + deltaPressure ) );
+
+    dTotalStress_dPressure[0] = biotCoefficient;
+    dTotalStress_dPressure[1] = biotCoefficient;
+    dTotalStress_dPressure[2] = biotCoefficient;
```

Comment:
Regarding the negative sign in line 74, can we explain the positive sign here?

Reply:
Besides, do you want to replace the Biot coefficient by the Biot tensor?

Reply:
Yes, it is appropriate to have a minus sign here. This requires a sign change also here:
https://github.com/GEOSX/GEOSX/blob/6ab47177cbbebe1eefb6553da7ebf03d4dea8c0e/src/coreComponents/physicsSolvers/multiphysics/SinglePhasePoromechanicsKernel.hpp#L281-L291
(Likewise in `MultiphasePoromechanicsKernel`).

Reply:
Done